### PR TITLE
Desktop: Fixes alt text not appearing in html

### LINKED ISSUE #5803: [Renderer] Alt-text of a local image is not rendered
```markdown
![this is alt text](:/06f049df282b4780bbfbdc306383bab5)
```
Its renderring result won't contain alt text. Is it a bug or intended feature? In fact, I''m a plugin developer and want to store some information as images' alt text. It would be better if alt text won't be ignored.

### PR DESCRIPTION
**Pull request by mak2002**
- Desktop: Fixes #5803 : Added alt text in images' html

Fixes #5803 : After adding images in Joplin the HTML was not showing alt text. 

For testing I added an image with alt text as nature-image.jpg.
added-alt-text2

### CODE REVIEW

File: packages/renderer/MdToHtml/rules/image.ts [hunk 1]
```diff
@@ -11,9 +11,9 @@ function plugin(markdownIt: any, ruleOptions: RuleOptions) {
 		const token = tokens[idx];
 		const src = utils.getAttr(token.attrs, 'src');
 		const title = utils.getAttr(token.attrs, 'title');
+		const alt = token.content;
```
Comment (laurent22) [inline packages/renderer/MdToHtml/rules/image.ts:14]: No need for the intermediate variable

File: packages/renderer/MdToHtml/rules/image.ts [hunk 2]
```diff
@@ -11,9 +11,9 @@ function plugin(markdownIt: any, ruleOptions: RuleOptions) {
...
 
 		if (!Resource.isResourceUrl(src) || ruleOptions.plainResourceRendering) return defaultRender(tokens, idx, options, env, self);
-
```
Comment (laurent22) [inline packages/renderer/MdToHtml/rules/image.ts:16]: no whitespace change please

File: packages/renderer/MdToHtml/rules/image.ts [hunk 3]
```diff
@@ -29,8 +29,7 @@ function plugin(markdownIt: any, ruleOptions: RuleOptions) {
 
 				js = ` ontouchstart="${touchStart}" ontouchend="${cancel}" ontouchcancel="${cancel}" ontouchmove="${cancel}"`;
 			}
-
-			return `<img data-from-md ${htmlUtils.attributesHtml(Object.assign({}, r, { title: title }))}${js}/>`;
+			return `<img data-from-md ${htmlUtils.attributesHtml(Object.assign({}, r, { title: title, alt: alt }))}${js}/>`;
```
Comment (laurent22) [inline packages/renderer/MdToHtml/rules/image.ts:32]: What happens when alt is not defined or an empty string?

**Review by laurent22 — COMMENTED** [review]
- Is the "content" part generally considered to be the "alt" attribute. Could you check the CommonMark spec and let us know please?

- Tests are missing

File: packages/renderer/MdToHtml/rules/image.ts (hunk 2 above)
Comment (mak2002) [reply packages/renderer/MdToHtml/rules/image.ts:16]: ok got it. I will take care from now on.

File: packages/renderer/MdToHtml/rules/image.ts (hunk 1 above)
Comment (mak2002) [reply packages/renderer/MdToHtml/rules/image.ts:14]: ok I will remove it and directly pass the value

**Comment by mak2002** [comment]
- @laurent22 I checked the CommonMark spec for [images](https://spec.commonmark.org/0.30/#images) and it says `An image description has inline elements as its contents. When an image is rendered to HTML, this is standardly used as the image’s alt attribute.`
- Soon I will add test cases and update the PR

File: packages/renderer/MdToHtml/rules/image.ts (hunk 3 above)
Comment (mak2002) [reply packages/renderer/MdToHtml/rules/image.ts:32]: When alt or title are not defined it returns ``

File: packages/renderer/MdToHtml/rules/image.ts [hunk 4]
```diff
@@ -11,9 +11,9 @@ function plugin(markdownIt: any, ruleOptions: RuleOptions) {
 		const token = tokens[idx];
 		const src = utils.getAttr(token.attrs, 'src');
 		const title = utils.getAttr(token.attrs, 'title');
-		const alt = token.content;
 
 		if (!Resource.isResourceUrl(src) || ruleOptions.plainResourceRendering) return defaultRender(tokens, idx, options, env, self);
+
```
Comment (mak2002) [inline packages/renderer/MdToHtml/rules/image.ts:16]: In my previous commit I accidentally removed a white space so this is just to make it like original.

**Comment by laurent22** [comment]
That looks good now, and thanks a lot for adding the tests. Let's wait till they have finished running on CI and then we can merge.

**Comment by mak2002** [comment]
@laurent22 Wow this will be my first code related PR in Joplin. Super happy and excited to contribute to Joplin. Thanks for helping me out.

**Comment by laurent22** [comment]
All tests are passing, thanks again for the PR!

**Pull request merged**